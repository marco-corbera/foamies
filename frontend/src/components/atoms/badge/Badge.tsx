import React from "react";

interface BadgeProps {
  text: string;
  bgColor?: string;
  className?: string;
}

const Badge: React.FC<BadgeProps> = ({
  text,
  bgColor = "bg-secondary",
  className = "",
}) => {
  return (
    <span
      className={`inline-block px-3 py-1 text-sm font-medium text-white rounded-full ${bgColor} ${className}`}
    >
      {text}
    </span>
  );
};

export default Badge;
