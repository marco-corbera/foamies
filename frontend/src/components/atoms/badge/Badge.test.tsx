import { render, screen } from "@testing-library/react";
import Badge from "@/components/atoms/badge";

describe("Badge tests", () => {
  it("Matches snapshot", () => {
    const { asFragment } = render(<Badge text={"My great badge"} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the badge", () => {
    render(<Badge text={"My great badge"} />);
    expect(screen.getByText("My great badge")).toBeInTheDocument();
  });
});
