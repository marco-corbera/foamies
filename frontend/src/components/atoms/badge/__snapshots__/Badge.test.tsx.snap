// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge tests Matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="inline-block px-3 py-1 text-sm font-medium text-white rounded-full bg-secondary-500"
  >
    My great badge
  </span>
</DocumentFragment>
`;
